hbase-client/src/main/java/org/apache/hadoop/hbase/zookeeper/ReadOnlyZKClient.java

```diff
@@ -253,6 +253,18 @@ public void closed(IOException e) {
     }
   }
 
+  public CompletableFuture<byte[]> getWithTimeout(String path, long timeout) {
+    CompletableFuture<byte[]> future = get(path);
+    while(timeout > 0){
+      if(future.isCancelled() || future.isDone() || future.isCompletedExceptionally()){
+        return future;
+      }
+      --timeout;
+    }
+    future.completeExceptionally(new KeeperException.OperationTimeoutException());
+    return future;
+  }
+
```

Comment:
@Divneet18 we need async implementation like mentioned here https://github.com/apache/hbase/pull/5837#issuecomment-2146135324